js/modules/loki_public_chat_api.js

```diff
@@ -691,6 +778,21 @@ class LokiPublicChannelAPI {
         }
       }
     }
+    const privKey = await this.serverAPI.chatAPI.getPrivateKey();
+    const ObjToSign = {
+      version: 1,
+      text,
+      annotations: payload.annotations,
+    };
+    if (payload.reply_to) {
+      ObjToSign.reply_to = payload.reply_to;
+    }
+    const sig = await libsignal.Curve.async.calculateSignature(
+      privKey,
+      JSON.stringify(ObjToSign)
+    );
+    payload.annotations[0].value.sig = StringView.arrayBufferToHex(sig);
+    payload.annotations[0].value.sigver = 1;
```

Comment:
Use `ObjToSign.version` ?

Reply:
it's not the version of the `network.loki.messenger.publicChat` annotation, it's the version of the signature format is my thinking. You could infer the signature version from the annotation version I suppose, but this current way gives up a bit more freedom and precision.

Reply:
Sorry if my question wasn't clear.
Basically there are two locations where we set a version to `1`, and to me it looks like it's the same value being repeated:
On the receiving/verification end, we use `annotation[0].value.sigver` to create `verifyObj.version`.
So I assumed that here on the sending end,
```js
const ObjToSign = {
      version: 1,
      ...
```
and
```js
payload.annotations[0].value.sigver = 1;
```
should be the same value too? If it's the case, let's put that into a constant global variable `SIGNATURE_VERSION` or something :)

Reply:
ah makes sense

Reply:
I think `objToSign.version` is the right move here. The version is closely tied to the code path, we may need to be able to write to multiple versions in the future, so a singular constant at the top doesn't make much sense to me.